server/catgenome/src/main/java/com/epam/catgenome/manager/FeatureIndexManager.java

```diff
@@ -344,6 +356,26 @@ public int getTotalPagesCount(VcfFilterForm filterForm, long projectId) throws I
         }
     }
 
+    private IndexSearchResult<FeatureIndexEntry> getGeneSearchResult(GeneFilterForm filterForm,
+                                                                     List<FeatureFile> featureFiles)
+            throws IOException {
+        if (filterForm.getPageSize() != null) {
+            LuceneIndexSearcher<FeatureIndexEntry> indexSearcher =
+                    getIndexSearcher(filterForm, featureIndexDao,
```

Comment:
I think that we should extract vcfManager from this method, now it is little bit confusing why I need to pash vcfManager to search Genes. 
To be able to do this we can refactor `AbstractIndexSearcher` and move `createSorting()` out from this class, in this case we will get rid of vcfManager

Reply:
Does ordering works now for gene? It seems to me that if we will specify ordering for sorting result - it will fail because it will try lo load vcf files with gene file ids, and as I understand there is no test case for this

Reply:
Done as you recommended. Also there will be no more conflicts about loading vcf files with gene ids - I have changed the code and add some new test cases.